Revert "Extended Legend testing"

eselmeister opened the pull request:
Reverts eclipse/swtchart#141

eselmeister committed:
Revert "Extended Legend testing"

eselmeister commented:
Dependency cycles occur ... I'll inspect this.

yashTEF commented:
Ok, I had switched dependency cycle errors as warnings in my workspace.


On Wed, 8 Apr 2020 at 15:25, eselmeister <notifications@github.com> wrote:

> Dependency cycles occur ... I'll inspect this.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610865369>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM5QYZTGIN6N5FVJAUNWMZ3RLRCYXANCNFSM4MDZNH6Q>
> .
>

eselmeister commented:
Hi Yash,

I have fixed it and also fixed also the NPE. I will upload the the fix soon.


Best,
Philip

Am 08.04.20 um 12:33 schrieb yashTEF:
> Ok, I had switched dependency cycle errors as warnings in my workspace.
>
>
> On Wed, 8 Apr 2020 at 15:25, eselmeister <notifications@github.com> wrote:
>
> > Dependency cycles occur ... I'll inspect this.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > 
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610865369>, or
> > unsubscribe
> > 
> <https://github.com/notifications/unsubscribe-auth/AM5QYZTGIN6N5FVJAUNWMZ3RLRCYXANCNFSM4MDZNH6Q>
> > .
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub 
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610881497>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAFHUTYINKX3OS2BLPCHS33RLRHGTANCNFSM4MDZNH6Q>.
>

-- 
~~~~~~~~~~~~~~~~~~~~~~~~
OpenChrom - the open source alternative for chromatography / mass spectrometry
Dr. Philip Wenig » Founder » philip.wenig@openchrom.net » http://www.openchrom.net
~~~~~~~~~~~~~~~~~~~~~~~~

eselmeister commented:
Hi Yash,

your demo went into "org.eclipse.swtchart.extensions" instead of 
"org.eclipse.swtchart.extensions.examples".
I have reverted the PR, added your demo and fixed the NPEs that have 
been introduced by the extended Legend.

Please have a look at the changes:
https://github.com/eclipse/swtchart/issues/143


Best,
Philip

Am 08.04.20 um 12:33 schrieb yashTEF:
> Ok, I had switched dependency cycle errors as warnings in my workspace.
>
>
> On Wed, 8 Apr 2020 at 15:25, eselmeister <notifications@github.com> wrote:
>
> > Dependency cycles occur ... I'll inspect this.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > 
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610865369>, or
> > unsubscribe
> > 
> <https://github.com/notifications/unsubscribe-auth/AM5QYZTGIN6N5FVJAUNWMZ3RLRCYXANCNFSM4MDZNH6Q>
> > .
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub 
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610881497>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAFHUTYINKX3OS2BLPCHS33RLRHGTANCNFSM4MDZNH6Q>.
>

-- 
~~~~~~~~~~~~~~~~~~~~~~~~
OpenChrom - the open source alternative for chromatography / mass spectrometry
Dr. Philip Wenig » Founder » philip.wenig@openchrom.net » http://www.openchrom.net
~~~~~~~~~~~~~~~~~~~~~~~~

eselmeister commented:
Hi Yash,

I made some additional improvements. Have a look at the latest commit:
https://github.com/eclipse/swtchart/issues/144


Best,
Philip

Am 08.04.20 um 12:33 schrieb yashTEF:
> Ok, I had switched dependency cycle errors as warnings in my workspace.
>
>
> On Wed, 8 Apr 2020 at 15:25, eselmeister <notifications@github.com> wrote:
>
> > Dependency cycles occur ... I'll inspect this.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > 
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610865369>, or
> > unsubscribe
> > 
> <https://github.com/notifications/unsubscribe-auth/AM5QYZTGIN6N5FVJAUNWMZ3RLRCYXANCNFSM4MDZNH6Q>
> > .
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub 
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610881497>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAFHUTYINKX3OS2BLPCHS33RLRHGTANCNFSM4MDZNH6Q>.
>

-- 
~~~~~~~~~~~~~~~~~~~~~~~~
OpenChrom - the open source alternative for chromatography / mass spectrometry
Dr. Philip Wenig » Founder » philip.wenig@openchrom.net » http://www.openchrom.net
~~~~~~~~~~~~~~~~~~~~~~~~

yashTEF commented:
Looks great,
Also could you tell me how to disable series plotting I am currently using
hideSeries but it still shows the series labels
Thanks,
Yash

On Wed, 8 Apr 2020, 18:13 eselmeister, <notifications@github.com> wrote:

> Hi Yash,
>
> I made some additional improvements. Have a look at the latest commit:
> https://github.com/eclipse/swtchart/issues/144
>
>
> Best,
> Philip
>
> Am 08.04.20 um 12:33 schrieb yashTEF:
> > Ok, I had switched dependency cycle errors as warnings in my workspace.
> >
> >
> > On Wed, 8 Apr 2020 at 15:25, eselmeister <notifications@github.com>
> wrote:
> >
> > > Dependency cycles occur ... I'll inspect this.
> > >
> > > —
> > > You are receiving this because you are subscribed to this thread.
> > > Reply to this email directly, view it on GitHub
> > >
> > <https://github.com/eclipse/swtchart/pull/142#issuecomment-610865369>,
> or
> > > unsubscribe
> > >
> > <
> https://github.com/notifications/unsubscribe-auth/AM5QYZTGIN6N5FVJAUNWMZ3RLRCYXANCNFSM4MDZNH6Q
> >
> > > .
> > >
> >
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/eclipse/swtchart/pull/142#issuecomment-610881497>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAFHUTYINKX3OS2BLPCHS33RLRHGTANCNFSM4MDZNH6Q
> >.
> >
>
> --
> ~~~~~~~~~~~~~~~~~~~~~~~~
> OpenChrom - the open source alternative for chromatography / mass
> spectrometry
> Dr. Philip Wenig » Founder » philip.wenig@openchrom.net »
> http://www.openchrom.net
> ~~~~~~~~~~~~~~~~~~~~~~~~
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610936423>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM5QYZTIPLOAOQ4MYGITIR3RLRWODANCNFSM4MDZNH6Q>
> .
>

eselmeister commented:
Hi Yash,

I have fixed the code, so that it can be also used in plain Java 
environments:
https://github.com/eclipse/swtchart/issues/145

Have a look at the extended chart legend how to display/hide a specific 
series, see attached screenshot.


Best,
Philip

Am 08.04.20 um 14:59 schrieb yashTEF:
> Looks great,
> Also could you tell me how to disable series plotting I am currently using
> hideSeries but it still shows the series labels
> Thanks,
> Yash
>
> On Wed, 8 Apr 2020, 18:13 eselmeister, <notifications@github.com> wrote:
>
> > Hi Yash,
> >
> > I made some additional improvements. Have a look at the latest commit:
> > https://github.com/eclipse/swtchart/issues/144
> >
> >
> > Best,
> > Philip
> >
> > Am 08.04.20 um 12:33 schrieb yashTEF:
> > > Ok, I had switched dependency cycle errors as warnings in my 
> workspace.
> > >
> > >
> > > On Wed, 8 Apr 2020 at 15:25, eselmeister <notifications@github.com>
> > wrote:
> > >
> > > > Dependency cycles occur ... I'll inspect this.
> > > >
> > > > —
> > > > You are receiving this because you are subscribed to this thread.
> > > > Reply to this email directly, view it on GitHub
> > > >
> > > <https://github.com/eclipse/swtchart/pull/142#issuecomment-610865369>,
> > or
> > > > unsubscribe
> > > >
> > > <
> > 
> https://github.com/notifications/unsubscribe-auth/AM5QYZTGIN6N5FVJAUNWMZ3RLRCYXANCNFSM4MDZNH6Q
> > >
> > > > .
> > > >
> > >
> > > —
> > > You are receiving this because you modified the open/close state.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/eclipse/swtchart/pull/142#issuecomment-610881497>,
> > > or unsubscribe
> > > <
> > 
> https://github.com/notifications/unsubscribe-auth/AAFHUTYINKX3OS2BLPCHS33RLRHGTANCNFSM4MDZNH6Q
> > >.
> > >
> >
> > --
> > ~~~~~~~~~~~~~~~~~~~~~~~~
> > OpenChrom - the open source alternative for chromatography / mass
> > spectrometry
> > Dr. Philip Wenig » Founder » philip.wenig@openchrom.net »
> > http://www.openchrom.net
> > ~~~~~~~~~~~~~~~~~~~~~~~~
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > 
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610936423>, or
> > unsubscribe
> > 
> <https://github.com/notifications/unsubscribe-auth/AM5QYZTIPLOAOQ4MYGITIR3RLRWODANCNFSM4MDZNH6Q>
> > .
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub 
> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610944478>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAFHUT4P4XERMOKAFBZDXJTRLRYLDANCNFSM4MDZNH6Q>.
>

-- 
~~~~~~~~~~~~~~~~~~~~~~~~
OpenChrom - the open source alternative for chromatography / mass spectrometry
Dr. Philip Wenig » Founder » philip.wenig@openchrom.net » http://www.openchrom.net
~~~~~~~~~~~~~~~~~~~~~~~~

eselmeister commented:
Hi Yash,

generally, SWTChart consists of two parts:

SWTChart Core
SWTChart Extensions

In the core, the charts / series can be modified directly, e.g. set 
background of the chart or the color/visibility of a series. In the 
extensions, these settings are encapsulated in setting classes 
IChartSettings, ISeriesSettings. These setting instances can be handled 
separately and are applied on the chart/series, when modifying it.

The base chart offers a function to suspend drawing the series. This 
could be used to draw the background image, e.g.:

baseChart.suspendUpdate(true);
create image and set it as the background
set series to status visible = false
baseChart.suspendUpdate(false);
baseChart.redraw();
...
let the user make the selection
...
baseChart.suspendUpdate(true);
remove the background image
set series to status visible = true
baseChart.suspendUpdate(false);
baseChart.redraw();

Just play a bit with the code. Let me know, if there are any uncertainties.


Best,
Philip


Am 08.04.20 um 22:43 schrieb Philip Wenig:
> Hi Yash,
>
> I have fixed the code, so that it can be also used in plain Java 
> environments:
> https://github.com/eclipse/swtchart/issues/145
>
> Have a look at the extended chart legend how to display/hide a 
> specific series, see attached screenshot.
>
>
> Best,
> Philip
>
> Am 08.04.20 um 14:59 schrieb yashTEF:
>> Looks great,
>> Also could you tell me how to disable series plotting I am currently 
>> using
>> hideSeries but it still shows the series labels
>> Thanks,
>> Yash
>>
>> On Wed, 8 Apr 2020, 18:13 eselmeister, <notifications@github.com> wrote:
>>
>> > Hi Yash,
>> >
>> > I made some additional improvements. Have a look at the latest commit:
>> > https://github.com/eclipse/swtchart/issues/144
>> >
>> >
>> > Best,
>> > Philip
>> >
>> > Am 08.04.20 um 12:33 schrieb yashTEF:
>> > > Ok, I had switched dependency cycle errors as warnings in my 
>> workspace.
>> > >
>> > >
>> > > On Wed, 8 Apr 2020 at 15:25, eselmeister <notifications@github.com>
>> > wrote:
>> > >
>> > > > Dependency cycles occur ... I'll inspect this.
>> > > >
>> > > > —
>> > > > You are receiving this because you are subscribed to this thread.
>> > > > Reply to this email directly, view it on GitHub
>> > > >
>> > > 
>> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610865369>,
>> > or
>> > > > unsubscribe
>> > > >
>> > > <
>> > 
>> https://github.com/notifications/unsubscribe-auth/AM5QYZTGIN6N5FVJAUNWMZ3RLRCYXANCNFSM4MDZNH6Q
>> > >
>> > > > .
>> > > >
>> > >
>> > > —
>> > > You are receiving this because you modified the open/close state.
>> > > Reply to this email directly, view it on GitHub
>> > > 
>> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610881497>,
>> > > or unsubscribe
>> > > <
>> > 
>> https://github.com/notifications/unsubscribe-auth/AAFHUTYINKX3OS2BLPCHS33RLRHGTANCNFSM4MDZNH6Q
>> > >.
>> > >
>> >
>> > --
>> > ~~~~~~~~~~~~~~~~~~~~~~~~
>> > OpenChrom - the open source alternative for chromatography / mass
>> > spectrometry
>> > Dr. Philip Wenig » Founder » philip.wenig@openchrom.net »
>> > http://www.openchrom.net
>> > ~~~~~~~~~~~~~~~~~~~~~~~~
>> >
>> > —
>> > You are receiving this because you commented.
>> > Reply to this email directly, view it on GitHub
>> > 
>> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610936423>, or
>> > unsubscribe
>> > 
>> <https://github.com/notifications/unsubscribe-auth/AM5QYZTIPLOAOQ4MYGITIR3RLRWODANCNFSM4MDZNH6Q>
>> > .
>> >
>>
>> —
>> You are receiving this because you modified the open/close state.
>> Reply to this email directly, view it on GitHub 
>> <https://github.com/eclipse/swtchart/pull/142#issuecomment-610944478>, 
>> or unsubscribe 
>> <https://github.com/notifications/unsubscribe-auth/AAFHUT4P4XERMOKAFBZDXJTRLRYLDANCNFSM4MDZNH6Q>.
>>
>
> -- 
> ~~~~~~~~~~~~~~~~~~~~~~~~
> OpenChrom - the open source alternative for chromatography / mass spectrometry
> Dr. Philip Wenig » Founder »philip.wenig@openchrom.net  »http://www.openchrom.net
> ~~~~~~~~~~~~~~~~~~~~~~~~

-- 
~~~~~~~~~~~~~~~~~~~~~~~~
OpenChrom - the open source alternative for chromatography / mass spectrometry
Dr. Philip Wenig » Founder » philip.wenig@openchrom.net » http://www.openchrom.net
~~~~~~~~~~~~~~~~~~~~~~~~
